autoemulate/experimental/compare.py

```diff
@@ -84,7 +86,11 @@ def compare(
             best_score_idx = scores.index(max(scores))
             best_model_config = configs[best_score_idx]
             cv_results = cross_validate(
-                cv(), self.train_val.dataset, model_cls, **best_model_config
+                cv=cv(),
+                dataset=self.train_val.dataset,
+                model=model_cls,
+                random_seed=self.random_seed,
```

Comment:
I'm wondering if the random seed is set in the `AutoEmulate `__init__()` (https://github.com/alan-turing-institute/autoemulate/pull/512#discussion_r2164555547) it can instead be passed here as `None`? Or perhaps it could be passed as a separate variable (e.g. `cv_seed`)?

Reply:
Not sure exactly what you mean here? If we passed it to the default CV model which is KFold that expects random_state so we could do:
`cv=cv(random_state=self.random_seed)`

Reply:
So yeah to add to this, I'm wondering if we should  distinguish between:
1. setting a random seed at the top-level `AutoEmulate` (in the init)
2. setting the random seed at the cross-validation level within the loop over models
for the `AutoEmulate` class?

I think for the compare loop it might make sense as default to set once and then pass `None` (or a variable like `cv_seed: int | None = None`) to the cross-validate call. This should be deterministic still at the `AutoEmulate` level providing all inner loops are deterministic.

But an option to make this not `None` and be able to set seed for the cross-validate call might be useful too (e.g. to fix model seed at this point and/or the cv splits too as you mention too in case it is useful to control the randomness at that level).

Reply:
Not sure I understand the reason for this? At the moment the only reason to set the random seed in the AutoEmulate init is for the specific purpose of passing it through to our cross_validate function in model_selection such that we call `set_random_seed(seed=random_seed)` right before fitting the model (and also pass the arg directly to RF/LGBM)